### PR TITLE
fixes #3460

### LINKED ISSUE #3460: Argument of type '{ name: FindOperator<any>; }' is not assignable to parameter of type 'DeepPartial<Entity>'
**Issue type:**

[x ] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.11`

Hello. My question is about https://github.com/typeorm/typeorm/blob/master/docs/find-options.md.
I can't use condition "In" at Entity.find method.

```
    Group.find{ name: In(["test"]) }

Argument of type '{ name: FindOperator<any>; }' is not assignable to parameter of type 'DeepPartial<Group>'.
  Types of property 'name' are incompatible.
    Type 'FindOperator<any>' is not assignable to type 'string'
```
but

```
getRepository(Group).find{ name: In(["test"]) }
still works
```

I compared Repository.find and BaseEntity.find methods interfaces and seen
1.    find(conditions?: FindConditions<Entity>): Promise<Entity[]>;
2.   static find<T extends BaseEntity>(this: ObjectType<T>, conditions?: DeepPartial<T>): Promise<T[]>;

What the reason this differences?

### PR DESCRIPTION
**Pull request by pleerock**

### CODE REVIEW

**Comment by pleerock** [comment]
It's breaking a bit since it adds type-safety to some methods (basically fixes their signatures).

File: src/entity-manager/EntityManager.ts [hunk 1]
```diff
@@ -682,7 +682,7 @@ export class EntityManager {
             throw new Error(`Value "${value}" is not a number.`);
 
         return this
-            .createQueryBuilder(entityClass, "entity")
+            .createQueryBuilder<any>(entityClass, "entity")
```
[the remark this answers is not in the document]
Comment (pleerock) [reply src/entity-manager/EntityManager.ts]: yeah I tried Entity and tsc gave me bunch of errors. Had no time to figure why.

[the remark this answers is not in the document]
Comment (Kononnable) [reply src/entity-manager/EntityManager.ts]: I think it should look more like that:
```typescript
return this
            .createQueryBuilder<Entity>(entityClass, "entity")
            .update(entityClass)
            .set({
                [propertyPath]: () => this.connection.driver.escape(column.databaseName) + " + " + value
            } as QueryDeepPartialEntity<Entity>)
            .where(conditions)
            .execute();
```

File: test/functional/query-builder/update/query-builder-update.ts [hunk 2]
```diff
@@ -238,7 +238,7 @@ describe("query builder > update", () => {
         try {
             await connection.createQueryBuilder()
                 .update(User)
-                .set({ unknownProp: true })
+                .set({ unknownProp: true } as any)
```
[the remark this answers is not in the document]
Comment (Kononnable) [reply test/functional/query-builder/update/query-builder-update.ts]: Ok, so if that's now checked on compilation time do we still need `EntityColumnNotFound`? We would just have to make sure every function returning new UpdateQueryBuilder use generic types the right way.

[the remark this answers is not in the document]
Comment (pleerock) [reply test/functional/query-builder/update/query-builder-update.ts]: > so if that's now checked on compilation time do we still need EntityColumnNotFound? 

yes we need in in runtime as well

File: src/entity-manager/EntityManager.ts [hunk 3]
```diff
@@ -708,7 +708,7 @@ export class EntityManager {
...
```
Comment (vlapo) [inline src/entity-manager/EntityManager.ts]: Use same construction as in `increment` method? `} as QueryDeepPartialEntity<Entity>`

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,12 +5,16 @@ however since API is already quite stable we don't expect too much breaking chan
...
-## Unreleased
+## 0.2.13 (unreleased)
 
 ### Bug Fixes
 
+* fixed signatures of `update`/`insert` methods, some `find*` methods in repositories, entity managers, BaseEntity and QueryBuilders
+
 ### Features
 
+## 0.2.12
```
Comment (vlapo) [inline CHANGELOG.md]: Why did you remove release date?

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,12 +5,16 @@ however since API is already quite stable we don't expect too much breaking chan
 If we missed a note on some change or you have a questions on migrating from old version,
 feel free to ask us and community.
 
-## Unreleased
+## 0.2.13 (unreleased)
 
 ### Bug Fixes
 
+* fixed signatures of `update`/`insert` methods, some `find*` methods in repositories, entity managers, BaseEntity and QueryBuilders
```
Comment (vlapo) [inline CHANGELOG.md]: What about link to issue after description? I know it is extra work but it is useful for users have a direct link to problem or change.

File: CHANGELOG.md (hunk 4 above)
Comment (pleerock) [reply CHANGELOG.md]: oups its a conflict resolve issue, we shouldn't have this line (see below line)

File: CHANGELOG.md (hunk 5 above)
Comment (pleerock) [reply CHANGELOG.md]: you are right, forgive me lazy 😭
